syntax = "proto3";

package ride;

import "google/protobuf/timestamp.proto";

message RideRequest {
    string passenger_id = 1;
    double from_lat = 2;
    double from_long = 3;
    double to_lat = 4;
    double to_long = 5;
    google.protobuf.Timestamp requested_at = 6;
}

enum RideStatus {
    REQUESTED = 0;
    ACCEPTED = 1;
    IN_PROGRESS = 2;
    COMPLETED = 3;
    CANCELED = 4;
}

message Ride {
    string ride_id = 1;
    RideRequest ride_request = 2;
    string driver_id = 3;
    float fare = 4;
    optional google.protobuf.Timestamp accepted_at = 5;
    optional google.protobuf.Timestamp finished_at = 6;
    RideStatus status = 7;
}

message RideInfo {
    Ride ride = 1;
    string passenger_name = 2;
    float current_lat = 3;
    float current_long = 4;
}

message Position {
    string ride_id = 1;
    float lat = 2;
    float long = 3;
    google.protobuf.Timestamp updated_at = 4;
}